src/NUnitFramework/framework/Constraints/CountZeroConstraint.cs

```diff
@@ -0,0 +1,63 @@
+// ***********************************************************************
+// Copyright (c) 2020 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+
+namespace NUnit.Framework.Constraints
+{
+    /// <summary>
+    /// CountZeroConstraint tests whether an instance has a property .Count with value zero.
+    /// </summary>
+    public class CountZeroConstraint : Constraint
+    {
+        private const string CountPropertyName = "Count";
+
+        /// <summary>
+        /// The Description of what this constraint tests, for
+        /// use in messages and in the ConstraintResult.
+        /// </summary>
+        public override string Description
+        {
+            get { return "<empty>"; }
+        }
+
+        /// <summary>
+        /// Checks if the specified <paramref name="type"/> has a Count property.
+        /// </summary>
+        /// <param name="type">Type to check.</param>
+        /// <returns><see langword="true"/> when <paramref name="type"/> has a 'Count' property, <see langword="false"/> otherwise.</returns>
+        public static bool HasCountProperty(Type type) => type.GetProperty(CountPropertyName) != null;
```

Comment:
Should we also validate if the type of the property is an `int`. Otherwise I think the cast `ApplyTo` would fail.

Reply:
Added a check on `PropertyType` and added new unit test. Initially I contemplated allowing more types, but I don't think it is worth it.